cmd/fetchd/cmd/genreplace.go

```diff
@@ -0,0 +1,164 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/server"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+)
+
+const (
+	flagNewDescription = "new-description"
+)
+
+// ReplaceGenesisValuesCmd returns replace-genesis-values cobra Command.
+func ReplaceGenesisValuesCmd(defaultNodeHome string) *cobra.Command {
```

Comment:
I would rename this to `ASIGenesisUpgrade(...)`.
This function will juts crate dedicated command with `asi-genesis-upgrade` name, and then inside of it's lambda function it would call other functions, each dedicated to specific tasks, like
* `ASIGenesisUpgradeReplaceDenom(...)`,
* `ASIGenesisUpgradeReplaceAddresses(...)`,
* `ASIGenesisUpgradeReplaceChainID(...)`,
* `ASIGenesisUpgradeWithdrawIBCChanelsBalances(...)`
* `ASIGenesisUpgradeWithdrawReconciliationWallets(...)`